File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/CommonUiObject.vb
Imports Application = System.Windows.Forms.Application
Imports System.Windows.Controls
Imports System.Windows.Media
Imports cadwiki.NetUtils
Imports cadwiki.NUnitTestRunner.Results
Imports System.Windows.Forms

Public Class CommonUiObject
    ReadOnly converter As BrushConverter = New BrushConverter()
    Public ReadOnly Green As Brush = CType(converter.ConvertFromString("#00FF00"), Brush)
    Public ReadOnly Red As Brush = CType(converter.ConvertFromString("#FF0000"), Brush)



    Public Sub AddWinformsTreeViewItemForTestResult(testResult As TestResult,
                                                    treeView As Windows.Forms.TreeView)
        treeView.BeginUpdate()
        Dim testNode As TreeNode = treeView.Nodes.Add("testResult.TestName")
        If testResult.Passed Then
            testNode.Nodes.Add("Passed: " + testResult.TestName)
            testNode.BackColor = System.Drawing.Color.Green
        Else
            testNode.Nodes.Add("Failed: " + testResult.TestName)
            testNode.Nodes.Add("Exception: " + testResult.ExceptionMessage)
            Dim stackTraceString As String = Lists.StringListToString(testResult.StackTrace, vbLf)
            testNode.Nodes.Add("Stack trace: " + stackTraceString)
            testNode.BackColor = System.Drawing.Color.Red
        End If
        treeView.EndUpdate()
        Application.DoEvents()
    End Sub

    Public Sub AddTreeViewItemForTestResult(testResult As TestResult,
                                            treeView As Windows.Controls.TreeView)
        Dim tvi As TreeViewItem = New TreeViewItem()
        tvi.Header = testResult.TestName
        If testResult.Passed Then
            tvi.Items.Add("Passed: " + testResult.TestName)
            tvi.Background = Green
        Else
            tvi.Items.Add("Failed: " + testResult.TestName)
            tvi.Background = Red
            tvi.Items.Add("Exception: " + testResult.ExceptionMessage)
            Dim stackTraceString As String = Lists.StringListToString(testResult.StackTrace, vbLf)
            tvi.Items.Add("Stack trace: " + stackTraceString)
        End If
        treeView.Items.Add(tvi)
        treeView.Items.Refresh()
        Application.DoEvents()
    End Sub

    Public Sub WpfAddResultsToTreeView(observableResults As ObservableTestSuiteResults,
                                   treeView As Windows.Controls.TreeView)
        Dim tvi As TreeViewItem = WpfCreateResultsItem(observableResults)
        treeView.Items.Add(tvi)
    End Sub

    Public Sub WpfUpdateResultsToTreeView(observableResults As ObservableTestSuiteResults,
                                      treeView As Windows.Controls.TreeView)
        Dim tvi As TreeViewItem = WpfCreateResultsItem(observableResults)
        treeView.Items.Item(0) = tvi
    End Sub

    Private Function WpfCreateResultsItem(
            observableResults As ObservableTestSuiteResults) As TreeViewItem
        Dim tvi As TreeViewItem = New TreeViewItem()
        tvi.Header = "Test Run Results: " + observableResults.TimeElapsed
        tvi.Items.Add("Total: " + observableResults.TotalTests.ToString())
        tvi.Items.Add("Passed: " + observableResults.PassedTests.ToString())
        tvi.Items.Add("Failed: " + observableResults.FailedTests.ToString())
        tvi.Items.Add("Time Elapsed: " + observableResults.TimeElapsed)
        tvi.IsExpanded = True
        Return tvi
    End Function

End Class

File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/CommonUiObject.cs
using System.Windows.Controls;
using System.Windows.Forms;
using Application = System.Windows.Forms.Application;
using System.Windows.Media;
using cadwiki.NetUtils;
using cadwiki.NUnitTestRunner.Results;
using Microsoft.VisualBasic;
using System;

namespace cadwiki.NUnitTestRunner
{

    public class CommonUiObject
    {
        private readonly BrushConverter converter = new BrushConverter();
        public readonly Brush Green;
        public readonly Brush Red;

        public CommonUiObject()
        {
            Green = (Brush)converter.ConvertFromString("#00FF00");
            Red = (Brush)converter.ConvertFromString("#FF0000");
        }



        public void WinFormsAddTreeViewItemForTestResult(TestResult testResult, System.Windows.Forms.TreeView treeView)
        {
            treeView.BeginUpdate();
            var testNode = treeView.Nodes.Add("testResult.TestName");
            if (testResult.Passed)
            {
                testNode.Nodes.Add("Passed: " + testResult.TestName);
                testNode.BackColor = System.Drawing.Color.Green;
            }
            else
            {
                testNode.Nodes.Add("Failed: " + testResult.TestName);
                testNode.Nodes.Add("Exception: " + testResult.ExceptionMessage);
                string stackTraceString = Lists.StringListToString(testResult.StackTrace, Environment.NewLine);
                testNode.Nodes.Add("Stack trace: " + stackTraceString);
                testNode.BackColor = System.Drawing.Color.Red;
            }
            treeView.EndUpdate();
            Application.DoEvents();
        }

        public void WinFormsAddResultsToTreeView(ObservableTestSuiteResults observableResults, System.Windows.Forms.TreeView treeView)
        {
            var node = WinFormsCreateResultsItem(observableResults);
            treeView.Nodes.Add(node);
        }

        public void WinFormsUpdateResultsToTreeView(ObservableTestSuiteResults observableResults, System.Windows.Forms.TreeView treeView)
        {
            var node = WinFormsCreateResultsItem(observableResults);
            treeView.Nodes.RemoveAt(0);
            treeView.Nodes.Insert(0, node);
        }

        private TreeNode WinFormsCreateResultsItem(ObservableTestSuiteResults observableResults)
        {
            var node = new TreeNode();
            node.Text = "Test Run Results: " + observableResults.TimeElapsed;
            node.Nodes.Add("Total: " + observableResults.TotalTests.ToString());
            node.Nodes.Add("Passed: " + observableResults.PassedTests.ToString());
            node.Nodes.Add("Failed: " + observableResults.FailedTests.ToString());
            node.Nodes.Add("Time Elapsed: " + observableResults.TimeElapsed);
            node.Expand();
            return node;
        }

        public void WpfAddTreeViewItemForTestResult(TestResult testResult, System.Windows.Controls.TreeView treeView)
        {
            var tvi = new TreeViewItem();
            tvi.Header = testResult.TestName;
            if (testResult.Passed)
            {
                tvi.Items.Add("Passed: " + testResult.TestName);
                tvi.Background = Green;
            }
            else
            {
                tvi.Items.Add("Failed: " + testResult.TestName);
                tvi.Background = Red;
                tvi.Items.Add("Exception: " + testResult.ExceptionMessage);
                string stackTraceString = Lists.StringListToString(testResult.StackTrace, Environment.NewLine);
                tvi.Items.Add("Stack trace: " + stackTraceString);
            }
            treeView.Items.Add(tvi);
            treeView.Items.Refresh();
            Application.DoEvents();
        }

        public void WpfAddResultsToTreeView(ObservableTestSuiteResults observableResults, System.Windows.Controls.TreeView treeView)
        {
            var tvi = WpfCreateResultsItem(observableResults);
            treeView.Items.Add(tvi);
        }

        public void WpfUpdateResultsToTreeView(ObservableTestSuiteResults observableResults, System.Windows.Controls.TreeView treeView)
        {
            var tvi = WpfCreateResultsItem(observableResults);
            treeView.Items[0] = tvi;
        }

        private TreeViewItem WpfCreateResultsItem(ObservableTestSuiteResults observableResults)
        {
            var tvi = new TreeViewItem();
            tvi.Header = "Test Run Results: " + observableResults.TimeElapsed;
            tvi.Items.Add("Total: " + observableResults.TotalTests.ToString());
            tvi.Items.Add("Passed: " + observableResults.PassedTests.ToString());
            tvi.Items.Add("Failed: " + observableResults.FailedTests.ToString());
            tvi.Items.Add("Time Elapsed: " + observableResults.TimeElapsed);
            tvi.IsExpanded = true;
            return tvi;
        }

    }
}